src/App.tsx

```diff
@@ -1,204 +1,112 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import {
+  FC, MouseEvent, useEffect, useState,
+} from 'react';
+import { Todo, Filter } from './types';
+import {
+  Header, TodoList, Footer, Errors, UserWarning,
+} from './components';
+import { getTodos } from './api/todos';
+
+const USER_ID = 12107;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
```

Comment:
I don't think you need additional `null` state here.

```suggestion
  const [todos, setTodos] = useState<Todo[]>([]);
```